api/provider/aws/environment_scaler.go

```diff
@@ -1,14 +1,818 @@
+		"m4.10xlarge":  NewInstanceSpec(40, 160),
+		"m4.16xlarge":  NewInstanceSpec(64, 256),
+		"m3.medium":    NewInstanceSpec(1, 3.75),
+		"m3.large":     NewInstanceSpec(2, 7.5),
+		"m3.xlarge":    NewInstanceSpec(4, 15),
+		"m3.2xlarge":   NewInstanceSpec(8, 30),
+		"c5.large":     NewInstanceSpec(2, 4),
+		"c5.xlarge":    NewInstanceSpec(4, 8),
+		"c5.2xlarge":   NewInstanceSpec(8, 16),
+		"c5.4xlarge":   NewInstanceSpec(16, 32),
+		"c5.9xlarge":   NewInstanceSpec(36, 72),
+		"c5.18xlarge":  NewInstanceSpec(72, 144),
+		"c4.large":     NewInstanceSpec(2, 3.75),
+		"c4.xlarge":    NewInstanceSpec(4, 7.5),
+		"c4.2xlarge":   NewInstanceSpec(8, 15),
+		"c4.4xlarge":   NewInstanceSpec(16, 30),
+		"c4.8xlarge":   NewInstanceSpec(36, 60),
+		"c3.large":     NewInstanceSpec(2, 3.75),
+		"c3.xlarge":    NewInstanceSpec(4, 7.5),
+		"c3.2xlarge":   NewInstanceSpec(8, 15),
+		"c3.4xlarge":   NewInstanceSpec(16, 30),
+		"c3.8xlarge":   NewInstanceSpec(32, 60),
+		"x1.16large":   NewInstanceSpec(64, 976),
+		"x1.32xlarge":  NewInstanceSpec(128, 1952),
+		"x1e.xlarge":   NewInstanceSpec(4, 122),
+		"x1e.2xlarge":  NewInstanceSpec(8, 244),
+		"x1e.4xlarge":  NewInstanceSpec(16, 488),
+		"x1e.8xlarge":  NewInstanceSpec(32, 976),
+		"x1e.16xlarge": NewInstanceSpec(64, 1952),
+		"x1e.32xlarge": NewInstanceSpec(128, 3904),
+		"r4.large":     NewInstanceSpec(2, 15.25),
+		"r4.xlarge":    NewInstanceSpec(4, 30.5),
+		"r4.2xlarge":   NewInstanceSpec(8, 61),
+		"r4.4xlarge":   NewInstanceSpec(16, 122),
+		"r4.8xlarge":   NewInstanceSpec(32, 244),
+		"r4.16xlarge":  NewInstanceSpec(64, 488),
+		"r3.large":     NewInstanceSpec(2, 15.25),
+		"r3.xlarge":    NewInstanceSpec(4, 30.5),
+		"r3.2xlarge":   NewInstanceSpec(8, 61),
+		"r3.4xlarge":   NewInstanceSpec(16, 122),
+		"r3.8xlarge":   NewInstanceSpec(32, 244),
+		"p3.2xlarge":   NewInstanceSpec(8, 61),
+		"p3.8xlarge":   NewInstanceSpec(32, 244),
+		"p3.16xlarge":  NewInstanceSpec(64, 488),
+		"p2.xlarge":    NewInstanceSpec(4, 61),
+		"p2.8xlarge":   NewInstanceSpec(32, 488),
+		"p2.16xlarge":  NewInstanceSpec(64, 732),
+		"g3.4xlarge":   NewInstanceSpec(16, 122),
+		"g3.8xlarge":   NewInstanceSpec(32, 244),
+		"g3.16xlarge":  NewInstanceSpec(64, 488),
+		"f1.2xlarge":   NewInstanceSpec(8, 122),
+		"f1.16xlarge":  NewInstanceSpec(64, 976),
+		"h1.2xlarge":   NewInstanceSpec(8, 32),
+		"h1.4xlarge":   NewInstanceSpec(16, 64),
+		"h1.8xlarge":   NewInstanceSpec(32, 128),
+		"h1.16xlarge":  NewInstanceSpec(64, 256),
+		"i3.large":     NewInstanceSpec(2, 15.25),
+		"i3.xlarge":    NewInstanceSpec(4, 30.5),
+		"i3.2xlarge":   NewInstanceSpec(8, 61),
+		"i3.4xlarge":   NewInstanceSpec(16, 122),
+		"i3.8xlarge":   NewInstanceSpec(32, 244),
+		"i3.16xlarge":  NewInstanceSpec(64, 488),
+		"i3.metal":     NewInstanceSpec(0, 512),
+		"d2.xlarge":    NewInstanceSpec(4, 30.5),
+		"d2.2xlarge":   NewInstanceSpec(8, 61),
+		"d2.4xlarge":   NewInstanceSpec(16, 122),
+		"d2.8xlarge":   NewInstanceSpec(36, 244),
+	}
```

Comment:
A brief glance through a handful of AWS documentation didn't yield anything like this for me - I too wanted to avoid hardcoding these things, but after the not-in-depth search didn't readily provide a solution, I figured the rest of the scaler was more important. I'd definitely love to do some more solid research later, though, and not have this human dependency, though. ✅